src/fetching/putEventResponse.ts

```diff
@@ -1,21 +1,34 @@
 import apiUrl from '../utils/apiUrl';
 
-import { ZetkinEventResponse, ZetkinEventSignup, ZetkinMembership } from '../types/zetkin';
-
-export default async function putEventResponse({ eventId, orgId } : ZetkinEventSignup) : Promise<ZetkinEventResponse> {
-    const mUrl = apiUrl('/users/me/memberships');
-    const mRes = await fetch(mUrl);
-    const mData = await mRes.json();
-    const orgMembership = mData.data.find((m : ZetkinMembership ) => m.organization.id === orgId);
-
-    if (orgMembership) {
-        const eventUrl = apiUrl(`/orgs/${orgId}/actions/${eventId}/responses/${orgMembership.profile.id}`);
-        const eventRes = await fetch(eventUrl, {
-            method: 'PUT',
-        });
-        const eventData = await eventRes.json();
-        return eventData.data;
-    }
-
-    throw 'no membership';
+import { ZetkinEventResponse, ZetkinMembership } from '../types/zetkin';
+
+function defaultFetch(path : string, init? : RequestInit) {
+    const url = apiUrl(path);
+    return fetch(url, init);
+}
+
+interface MutationVariables {
+    eventId: number;
+    orgId: number;
+}
+
+export default function putEventResponse(fetch = defaultFetch) {
```

Comment:
Is `fetch` injection necessary for this function? Is it ever used on the server?

Reply:
The same question stands for the delete variant as well.

Reply:
That's true, it's only used from the client. I'll change this.